php/proxy.php

```diff
@@ -3,150 +3,192 @@
+                $headers[$name] = $value;
+            } else if ($name == "CONTENT_TYPE") {
+                $headers["Content-Type"] = $value;
+            } else if ($name == "CONTENT_LENGTH") {
+                $headers["Content-Length"] = $value;
+            }
+        }
+        return $headers;
+    }
+
     private function getHeadersOfRequest() {
         $headers = array();
-        $isAuthToken = false; 
-        foreach (getallheaders() as $name => $value) {
-          $headerString = $name.':'.$value;
-
-          //Setting target Server as Host
-          if(strtolower($name) == 'host') {
-              $headerString = $name.':'.parse_url(self::getSettings()['url'])['host'];
-          }
-
-          if($name == 'authToken') {
-            $isAuthToken = true;
-          }
-          array_push($headers,"$headerString");
+        $isAuthToken = false;
+
+        foreach (self::emulate_getallheaders() as $name => $value) {
+
+            $headerString = $name.':'.$value;
+
+            //Setting target Server as Host
+            if(strtolower($name) == 'host') {
+                $headerString = $name.':'.$this->host;
+            }
+            if($name == 'authToken') {
+                $isAuthToken = true;
+            }
+            array_push($headers,"$headerString");
         }
-        
+
         if($isAuthToken == false){
-            $settings = self::getSettings();
-            $username = 'username:'.$settings['username'];
-            $password = 'password:'.$settings['password'];
-            array_push($headers, $username);
-            array_push($headers, $password);
-            array_push($headers,'User-Agent:Tinymce Proxy');
+            $this->settings = self::getSettings();
+            $this->username = 'username:'.$this->settings['username'];
+            $this->password = 'password:'.$this->settings['password'];
+            array_push($headers, $this->username);
+            array_push($headers, $this->password);
+            array_push($headers,'User-Agent:Drupal Proxy');
+
         }
         return $headers;
     }
-    
+
     private function getPOSTdata(){
         return file_get_contents("php://input");
     }
-    
-    private function proxyRequest($url){
+
+    public function __construct() {
+        self::disable_gzip();
+        $this->settings = self::getSettings();
+        $this->targetPath = rtrim($this->settings['url'],"/");
+        $this->host = parse_url($this->targetPath)['host'];
+
+        if(isset($_SERVER['HTTPS'] ) && $_SERVER['HTTPS'] == 'on'){
+            $this->domainURL = 'https://'.$_SERVER['HTTP_HOST'];
+        } else {
+            $this->domainURL = 'http://'.$_SERVER['HTTP_HOST'];
+        }
+
+        $part = '/acrolinx/proxy.php'; //TODO: Ensure that this is the correct location of the 'proxy.php' on the server.
+        $requestURI = $_SERVER['REQUEST_URI'];
+        $pos = self::getPosition($requestURI, $part);
+
+        if ($pos !== false) {
```

Comment:
Oh we already do a check... But as result we will just stop working? Probably we should be able to deal with the situation when the path is not as expected or we should return some useful message back to the browser...

Reply:
FIx: Returning message to the browser to ensure correct location of proxy.php

Reply:
If we can detect - why does the integrator have to change the variable in line 105?

Reply:
If I understood your question completely and correctly,
What if Integrator wants to place proxy.php in some other directory than /acrolinx/ like www/myplugin/myproxy/proxy.php ?
If we are ok that integrator should place proxy.php always in /acrolinx/ directory ?
Let me know if I am missing something

Reply:
Hi @acrork 

$part = "/acrolinx/proxy.php" <-- /acrolinx is directory in which proxy.php is placed.

If integrator place proxy.php in some other directory like "abc" then /acrolinx/proxy.php will be invalid. 
In that case integrator will require to change /acrolinx/proxy.php to /abc/proxy.php 

The line of code if ($pos !== false) checks for part the location of $part for getting the actual request URI.

If the above explanation is confusing,
Could you explain how the proxy would work if it is placed at: C:\xampp\htdocs\abc\proxy.php
Would it just return error every time at if ($pos !== false)

Sorry if I am causing more confusion for the comment.

Reply:
Fixed: Used $_SERVER['PHP_SELF']; and set it to $part